_inc/lib/class.jetpack-service-helper.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+class Jetpack_Service_Helper {
+	/**
+	 * @var Jetpack_Service_Helper
+	 **/
+	private static $instance = null;
+
+	static function init() {
+		if ( is_null( self::$instance ) ) {
+			self::$instance = new Jetpack_Service_Helper;
+		}
+
+		return self::$instance;
+	}
+
+	private function __construct() {
+		add_action( 'load-settings_page_sharing', array( __CLASS__, 'admin_page_load' ), 9 );
+	}
+
+	/**
+	 * Gets a URL to the public-api actions. Works like WP's admin_url
+	 *
+	 * @param string $service Shortname of a specific service.
+	 *
+	 * @return URL to specific public-api process
+	 */
+	// on WordPress.com this is/calls Keyring::admin_url
+	static function api_url( $service = false, $params = array() ) {
+		/**
+		 * Filters the API URL used to interact with WordPress.com.
+		 *
+		 * @module publicize
+		 *
+		 * @since 2.0.0
+		 *
+		 * @param string https://public-api.wordpress.com/connect/?jetpack=publicize Default Publicize API URL.
+		 */
+		$url = apply_filters( 'publicize_api_url', 'https://public-api.wordpress.com/connect/?jetpack=publicize' );
```

Comment:
This calls the `public_api_keyring_connect` function on wpcom which is good, however it has plenty of special cases for publicize services. I think we could add a new value for `jetpack` here which would do the same thing on wpcom but stop after the keyring connection is created and not attempt to create a publicize connection or redirect some place else.

Reply:
This code is shared between the Publicize module and the Site Verification module, so I think right now this might be a little messy. I think we should backlog figuring out the right solution to another patch.

Reply:
@mdawaffe has recently introduced a new way to sandbox Jetpack using pre-request filters here: https://github.com/Automattic/jetpack/blob/master/_inc/jetpack-server-sandbox.php

Maybe it would be good to include the needed logic there after this PR is merged instead of adding it here.